core/src/tasks/profile/confirm.ts

```diff
@@ -0,0 +1,61 @@
+import Moment from "moment";
+import { CLSTask } from "../../classes/tasks/clsTask";
+import { CLS } from "../../modules/cls";
+import { plugin } from "../../modules/plugin";
+import { Schedule } from "../../models/Schedule";
+import { Run } from "../../models/Run";
+import { ProfileOps } from "../../modules/ops/profile";
+
+export class ProfilesConfirm extends CLSTask {
+  constructor() {
+    super();
+    this.name = "profiles:confirm";
+    this.description = "Confirm that profiles still exist in the source";
+    this.frequency = 1000 * 30;
```

Comment:
Nice!  So this one task will check both properties owned by a source's schedule (that's a mouthful) that are old AND any properties that are globally old.

Reply:
I'm a little curious about the processing time involved here: 
* This task runs every 30 seconds.  
* If there are a lot of Properties to update, this task will re-enqueue itself
* In that time, 30s may have passed, and this task will be re-enqueued again
* If the server is spending all it's time on  `ProfileOps.confirmExistence`, it's likely that none of the Imports could get processed... and we might have an loop?

While the `import` queue is higher in priority than the `profiles` queue, the properties might not be updated before the next loop as the `profileProperties` priority is after `profiles`

---

Update: I think I figured it out! The import will make the ProfileProperty not `ready`, and therefore not found in the next loop of `confirmExistence`.  Was that the intent @pedroslopez ?

Reply:
You got it! Affected profiles (and their properties) become `pending` and `confirmExistence` explicitly looks for profiles and properties that are `ready`. Once these properties go back to being `ready`, they'll have a new `confirmedAt` timestamp as well so `confirmExistence` again won't pick them up until it's time to do so.